### PR TITLE
DDO-775 Shared resources for Sam preview envs

### PR DESCRIPTION
**Pull request by gmalkov**

### CODE REVIEW

File: sam/dns.tf [hunk 1]
```diff
@@ -0,0 +1,21 @@
+data "google_dns_managed_zone" "dns_zone" {
+  count = var.enable && contains(["default", "preview"], var.env_type) ? 1 : 0
+
+  provider = google.dns
+  name     = var.dns_zone_name
+}
+
+locals {
+  fqdn = var.enable && contains(["default", "preview"], var.env_type) ? "${local.hostname}${local.subdomain_name}.${data.google_dns_managed_zone.dns_zone[0].dns_name}" : null
```
Comment (mikeflinn) [inline sam/dns.tf:9]: It's getting to be pretty tough to understand what the end result of the fqdn is at this point. I wonder if there's a more readable way we could express this.

File: sam/firestore.tf [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Google project for Sam Firestore
+resource "google_project" "sam-firestore" {
+  count = var.enable && contains(["default", "preview_shared"], var.env_type) ? 1 : 0
```
Comment (mikeflinn) [inline sam/firestore.tf:3]: Would it be easier to make a local for this expression since it is repeated alot?

**Review by mikeflinn — APPROVED** [review]
Looks great, left a few minor comments

File: sam/firestore.tf (hunk 2 above)
Comment (gmalkov) [reply sam/firestore.tf:3]: What's in the `contains` is different for different resources base don which types of envs they should be created in. I guess I can make locals for all of the different `var.enable && contains([***], var.env_type)` combinations? Not sure how much cleaner that is.

File: sam/dns.tf (hunk 1 above)
Comment (gmalkov) [reply sam/dns.tf:9]: Nothing obvious comes to mind but I can play around with it

File: sam/firestore.tf (hunk 2 above)
Comment (mikeflinn) [reply sam/firestore.tf:3]: Ah I thought they would all evaluate to the same thing within a particular invocation of the module. If that isn't the case then yeah I guess there isn't much benefit to doing this.

**Pull request merged**